kasa/cli.py

```diff
@@ -630,18 +631,113 @@ async def schedule(dev):
 
 @schedule.command(name="list")
 @pass_dev
+@click.option("--json", is_flag=True)
 @click.argument("type", default="schedule")
-def _schedule_list(dev, type):
+def _schedule_list(dev, json, type):
     """Return the list of schedule actions for the given type."""
     sched = dev.modules[type]
-    for rule in sched.rules:
-        print(rule)
-    else:
+    if sched.rules == []:
         echo(f"No rules of type {type}")
+    for rule in sched.rules:
+        if json:
+            print(rule.json())
+        else:
+            print(rule)
 
     return sched.rules
 
 
+@schedule.command(name="enable")
+@pass_dev
+@click.argument("enable", type=click.BOOL)
+async def _schedule_enable(dev, enable):
+    """Enable or disable schedule."""
+    schedule = dev.modules["schedule"]
+    return await schedule.set_enabled(1 if state else 0)
+
+
+@schedule.command(name="add")
+@pass_dev
+@click.option("--name", type=str, required=True)
+@click.option("--enable", type=click.BOOL, default=True, show_default=True)
+@click.option("--repeat", type=click.BOOL, default=True, show_default=True)
+@click.option("--wday", type=str, required=True)
+@click.option("--sact", type=click.IntRange(-1, 2), default=None, required=True)
+@click.option("--stime_opt", type=click.IntRange(-1, 2), default=None, required=True)
+@click.option("--smin", type=click.IntRange(0, 1440), default=None, required=True)
+@click.option("--eact", type=click.IntRange(-1, 2), default=-1)
+@click.option("--etime_opt", type=click.IntRange(-1, 2), default=-1)
+@click.option("--emin", type=click.IntRange(0, 1440), default=None)
```

Comment:
I think we should modiy the `Rule` to have more user-friendly field names (using [`alias`](https://docs.pydantic.dev/latest/usage/fields/#field-aliases)) and add some documentation on what input values they accept.

Reply:
I changed the names (If you better ones please comment), the doc still needs to be done.

Is there a way to pass the pydatic info to click? for example `TimeOption`

Reply:
I don't have a direct answer to that, sorry. I suppose generic support for all pydantic models is not that easy, but one can use manually inherit from [`click.ParamType`](https://click.palletsprojects.com/en/8.1.x/api/#click.ParamType) to perform the conversion.